internal/driver/driver_test.go

```diff
@@ -1023,6 +1041,96 @@ func TestSymbolzAfterMerge(t *testing.T) {
 	}
 }
 
+func TestHttpsInsecure(t *testing.T) {
+	baseVars := pprofVariables
+	pprofVariables = baseVars.makeCopy()
+	defer func() { pprofVariables = baseVars }()
+
+	tlsConfig := &tls.Config{Certificates: []tls.Certificate{selfSignedCert(t)}}
+	donec := make(chan struct{}, 1)
+
+	l, err := tls.Listen("tcp", "localhost:0", tlsConfig)
+	if err != nil {
+		t.Fatalf("net.Listen: got error %v, want no error", err)
+	}
+	defer func() { <-donec }()
+	defer l.Close()
+
+	go func(donec chan<- struct{}) {
+		defer func() { donec <- struct{}{} }()
+		if got, want := http.Serve(l, nil), "use of closed"; !strings.Contains(got.Error(), want) {
+			t.Fatalf("Serve got error %v, want %q", got, want)
+		}
+	}(donec)
+
+	go func() {
+		deadline := time.Now().Add(5 * time.Second)
+		for time.Now().Before(deadline) {
+			// Simulate a hotspot function.
+		}
+	}()
+
+	outputTempFile, err := ioutil.TempFile("", "profile_output")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer os.Remove(outputTempFile.Name())
+	defer outputTempFile.Close()
+
+	f := emptyFlags()
+	o := setDefaults(nil)
+	o.Flagset = &f
+
+	f.args = []string{"https+insecure://" + l.Addr().String() + "/debug/pprof/profile?seconds=5"}
+	addFlags(&f, []string{
+		"top",
+		"symbolize=remote",
+		"output=" + outputTempFile.Name(),
+	})
+
+	if err := PProf(o); err != nil {
+		t.Fatalf("PProf(%v): got error %v, want no error", o, err)
+	}
+
+	b, err := ioutil.ReadFile(outputTempFile.Name())
+	if err != nil {
+		t.Fatalf("ReadFile(%s) got error %v, want no error", outputTempFile.Name(), err)
+	}
+	if got, want := string(b), "TestHttpsInsecure"; !strings.Contains(got, want) {
```

Comment:
this test doesn't pass on 1.8:
```
--- FAIL: TestHttpsInsecure (5.03s)
	driver_test.go:1098: Pprof(&{{} 0xc4200157c0 <nil> 0xc420167f40 0xc42016a960 0xc42000e048}): got Type: cpu
		Time: Mar 14, 2017 at 6:39pm (EDT)
		Duration: 5s, Total samples = 4.36s (87.13%)
		Showing nodes accounting for 4.36s, 100% of 4.36s total
		      flat  flat%   sum%        cum   cum%
		     4.36s   100%   100%      4.36s   100%  <unknown>
		, want "TestHttpsInsecure" substring
FAIL
```

It does pass on Go 1.7, and the profile above just looks garbled, so I'm not sure what to make of it.